src/components/grader/submissions/SingleDiff.js

```diff
@@ -1,35 +1,134 @@
 import ReactDiffViewer from "react-diff-viewer";
-import React from "react";
-import AccordionSection from "@bit/nexxtway.react-rainbow.accordion-section";
-import { Label } from "semantic-ui-react";
+import React, { useState, useEffect } from "react";
+import { AccordionItem, Tile } from 'carbon-components-react';
+import CodeEditor from "./CodeEditor"
+import apiClient from "../../../api-client";
+
 
 function SingleDiff(props) {
+
+  const [theList, setTheList] = useState([])
+  const [visible, setVisible] = useState(false)
+
+
+  useEffect(() => {
+    apiClient.getFilesList(props.course, props.lab, props.result.id)
+      .then(res => {
+        setTheList(res.data)
+      });
+// eslint-disable-next-line
+  }, [])
+
+
+  useEffect(() => {
+   
+    if(theList.length > 0) {
+      setVisible(true)
+    }
+
+
+  }, [theList])
+
   return (
     <div>
-      <AccordionSection
-        label={props.result.id}
-        style={{ marginBottom: "-30px" }}
+      <AccordionItem
+        title={props.result.id}
       >
+        {
+          visible &&
+          <CodeEditor
+            subList={theList}
+            course={props.course}
+            lab={props.lab}
+            submissionId={props.result.id}
+          />
+        }
+
         {props.result.failed.map((res) => (
           <div key={res.tc_id} >
-            <Label as="a" color="red" image>
-              {res.tc_id}
-              <Label.Detail>Test Case Id</Label.Detail>
-            </Label>
-
+            <Tile> Test Case {res.tc_id}</Tile>
             <hr></hr>
-            <ReactDiffViewer leftTitle="Expected"
+
+            <ReactDiffViewer
+              leftTitle="Expected"
               rightTitle="Output"
               oldValue={res.expected}
               newValue={res.output}
+              useDarkTheme={true}
+            //styles={newStyles}
             />
-
             <br></br>
           </div>
         ))}
-      </AccordionSection>
+
+        {/* <CodeEditor subList={sub_list}
+            codeFiles={code_files}
+            defaultLabel={sub_list[0].submssionFile}
+          /> */}
+
+      </AccordionItem>
+
+
     </div>
   );
 }
 
 export default SingleDiff;
+
+
+// const newStyles = {
```

Comment:
do we need this code?

Reply:
just in case, we override the `DiffViewer` style